server/server/data_loader/dictionary_full.py

```diff
@@ -1,16 +1,51 @@
 import json
 from tqdm import tqdm
+from collections import Counter
+from .textfunctions import asciify_roman, pali_sort_key
+from .fuzzymatcher import FuzzyMatcher
 
-def load_dictionary_full(db, dictionaries_dir):
+def load_dictionary_full(db, dictionaries_dir, change_tracker):
     print('Loading dictionary_full')
-    dictionary_full = (dictionaries_dir / 'en').glob('*.json')
+    
+    dictionary_full_files = list((dictionaries_dir / 'en').glob('*.json'))
+    if not change_tracker.is_any_file_new_or_changed(dictionary_full_files):
+        return
+    
     dictionary_full_collection = db['dictionary_full']
-    db['dictionary_full'].truncate()
     docs = []
-    for dictionary in tqdm(dictionary_full):
-        with dictionary.open('r', encoding='utf-8') as f:
-
+    # For now hardcode this, in the future we may have other pairs
+    lang_from = 'pli'
+    lang_to = 'en'
+    words_seen = Counter()    
+    ids_seen = Counter()
+    for dictionary in tqdm(dictionary_full_files):
+        with dictionary.open('r', encoding='utf-8') as f:    
             for entry in json.load(f):
```

Comment:
if you do full data load anyway it is better to store that data into the variable, close the file and then do data manipulation outside with block. so:
```
with dictionary.open('r', encoding='utf-8') as f:
    data = json.load(f)
for...
```